src/store/auth.module.js

```diff
@@ -24,50 +24,50 @@ const getters = {
   }
 };
 
-const actions = {
-  [LOGIN](context, credentials) {
-    return new Promise(resolve => {
-      ApiService.post("users/login", { user: credentials })
-        .then(({ data }) => {
-          context.commit(SET_AUTH, data.user);
-          resolve(data);
-        })
-        .catch(({ response }) => {
-          context.commit(SET_ERROR, response.data.errors);
-        });
-    });
+export const actions = {
+  async [LOGIN](context, credentials) {
+    try {
+      const {
+        data: { user = {} }
```

Comment:
Can we keep Object destructuring at one level? Otherwise, it gets unreadable.

Reply:
I tend to quite like nested destructuring since it allows you to easily assign default values without having to do a separate check at each level and assign new variable names. I accept that it can make it a bit less readable for people not used to it though so I'll simplify it.